common/dbus/platform/asio_libdbus/dbus_client.cpp

```diff
@@ -447,6 +510,16 @@ DBusHandlerResult MsgFilter(DBusConnection *connection, DBusMessage *message, vo
 		auto ex_value = ExtractValueFromDBusMessage<expected::ExpectedString>(message);
 		(*opt_string_handler)(ex_value);
 		return DBUS_HANDLER_RESULT_HANDLED;
+	} else if (opt_string_pair_handler) {
```

Comment:
I was nearly expecting a templated function here also :laughing:

Reply:
If you find a way to do it that way, sure! I wasn't able to. But there's definitely nothing left to lose here. :sweat_smile: